### PR TITLE
new testing environment with testing database

### PR DESCRIPTION
**Pull request by karliatto**
- new test environment with a database for testing with automating deleting after test
- new tests adapted to new structure
- readme with explanation how to create testing database and how to run tests

### CODE REVIEW

**Comment by chill117** [comment]
Looks like there is an issue with the travis CI configuration:
```
Error Pool2 - Error: ER_ACCESS_DENIED_ERROR: Access denied for user 'bitstarterTest'@'localhost' (using password: YES)
```
See https://travis-ci.org/Learn-by-doing/bitstarter/jobs/179870423

The travis CI configuration file needs pre-install instructions to setup the test database before the tests are run:
https://github.com/Learn-by-doing/bitstarter/blob/master/.travis.yml

**Comment by karliatto** [comment]
I see,  I found some information :
https://docs.travis-ci.com/user/database-setup/
https://docs.travis-ci.com/user/languages/javascript-with-nodejs/

**Comment by chill117** [comment]
@karliatto Yes, I think those are the correct instructions.

Also, if you ever need to trigger the travis build just close and re-open the pull request.

File: .travis.yml [hunk 1]
```diff
@@ -16,3 +22,4 @@ addons:
 
 env:
   - CXX=g++-4.8
+  - DB_HOST="127.0.0.1" DB_PORT=3306 DB_USER="bitstarterTest" DB_PASS="bitstarterTest" DB_NAME="bitstarterTest"
```
Comment (chill117) [inline .travis.yml:25]: Each line under the `env:` section will start a new test environment in travis. Each of those environments will run all the tests. You must combine the environment variables here into one line, like this:
```yml
env:
   - CXX=g++-4.8 DB_HOST="127.0.0.1" DB_PORT=3306 DB_USER="bitstarterTest" DB_PASS="bitstarterTest" DB_NAME="bitstarterTest"
```

**Pull request closed without merging**